quarkus/extensions/src/main/java/org/keycloak/transaction/QuarkusJtaTransactionManagerLookup.java

```diff
@@ -32,16 +32,20 @@
 
     @Override
     public TransactionManager getTransactionManager() {
-        return tm;
+        if (tm == null) {
+//            tm = CDI.current().select(TransactionManager.class).get();
+//            logger.tracev("TransactionManager = {0}", tm);
+//            if (tm == null) {
+//                logger.debug("could not locate transactionmanager");
+//            }
+        }
+        // we need to check whether or not the datasource supports XA as well as if configuration tells that
+        return null;
     }
 
     @Override
     public void init(Config.Scope config) {
-        tm = CDI.current().select(TransactionManager.class).get();
-        logger.tracev("TransactionManager = {0}", tm);
-        if (tm == null) {
-            logger.debug("Could not locate TransactionManager");
-        }
+        
```

Comment:
I'm confused here. Don't we need at least a transaction manager dummy or something? Otherwise session.getTransactionManager won't work

Reply:
This is something I was expecting to investigate more once I have these changes in. It is not clear to me if we always need to run using JTA. Looks like it is not.